File: src/main/java/org/project/heredoggy/user/walk/walkRecord/dto/WalkRecordEndRequestDTO.java
package org.project.heredoggy.user.walk.walkRecord.dto;

import lombok.*;

import java.util.List;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class WalkRecordEndRequestDTO {
    private Double actualDistance;
    private Integer actualDuration;
    private List<WalkRecordPointDTO> actualPath;
}

File: src/main/frontend/lib/models/walk_record_end_request_dto.dart
import 'package:flutter/foundation.dart';
import 'walk_record_point_dto.dart';

class WalkRecordEndRequestDTO {
  final double actualDistance;
  final int actualDuration;
  final List<WalkRecordPointDTO> actualPath;

  WalkRecordEndRequestDTO({
    required this.actualDistance,
    required this.actualDuration,
    required this.actualPath,
  });

  Map<String, dynamic> toJson() => {
    'actualDistance': actualDistance,
    'actualDuration': actualDuration,
    'actualPath': actualPath.map((x) => x.toJson()).toList(),
  };
} 